internal/security/secretstore/config.go

```diff
@@ -0,0 +1,57 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ *******************************************************************************/
+
+package secretstore
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/edgexfoundry/edgex-go/internal/pkg/config"
+)
+
+type ConfigurationStruct struct {
+	Writable      WritableInfo
+	Logging       config.LoggingInfo
+	SecretService SecretServiceInfo
+}
+
+type WritableInfo struct {
+	LogLevel string
+	Title    string
+}
+
+type SecretServiceInfo struct {
+	Scheme               string
+	Server               string
+	Port                 int
+	CertPath             string
+	CaFilePath           string
```

Comment:
A little baffling
<foo>path, <foo>file
but we have a CertPath and CertFilePath (is this path plus file name?, then do we need CertPath?)
CaFilePath -- is this to a key. or the CA cert?

Reply:
I read it as `CertPath` is the path to look for system certs to verify the TLS host, while `CertFilePath` is a file path to a specific file that should be used to verify the TLS host.

Reply:
but agreed it's confusing and could use a rename

Reply:
I would like to keep it as is for consistence. If we decide to divide it into a couple of smaller modules, we will have chance to address them there.

Reply:
Create issue #1762  to track the incoming update.